kyo-data/shared/src/main/scala/kyo/Schedule.scala

```diff
@@ -0,0 +1,348 @@
+package kyo
+
+import Schedule.internal.*
+import kyo.Duration
+import kyo.Instant
+
+/** An immutable, composable scheduling policy.
+  *
+  * Schedule provides various combinators for creating complex scheduling policies. It can be used to define retry policies, periodic tasks,
+  * or any other time-based scheduling logic.
+  */
+sealed abstract class Schedule derives CanEqual:
```

Comment:
For consistency, how would you feel about `self =>`?

Reply:
I think in this case it wouldn't matter? There aren't nested classes where `this` would be ambiguous